base_layer/key_manager/src/key_manager_service/interface.rs

```diff
@@ -178,22 +178,19 @@ where PK: ByteArray
             None => Err("Out of bounds".to_string()),
             Some(val) => match *val {
                 MANAGED_KEY_BRANCH => {
-                    if parts.len() != 3 {
-                        return Err("Wrong managed format".to_string());
-                    }
-                    let index = parts[2]
+                    let index = parts
```

Comment:
We still need these checks, no? It is not only index checking.
```suggestion
                    if parts.len() != 3 {
                        return Err("Wrong managed format".to_string());
                    }
                    let index = parts
```